File: 03.CPP/0167.cpp
#include <vector>
#include <iostream>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& numbers, int target) {
        int left = 0;
        int right = numbers.size() - 1;

        while (left < right) {
            int currentSum = numbers[left] + numbers[right];

            if (currentSum < target) {
                left++;
            } else if (currentSum > target) {
                right--;
            } else {
                return {left + 1, right + 1};
            }
        }

        return {-1, -1};
    }
};

void printVector(const std::vector<int>& vec) {
    std::cout << "[";
    for (size_t i = 0; i < vec.size(); ++i) {
        std::cout << vec[i] << (i < vec.size() - 1 ? ", " : "");
    }
    std::cout << "]" << std::endl;
}

int main() {
    Solution sol;

    std::vector<int> nums = {2, 7, 11, 15};
    int target = 9;
    std::vector<int> result = sol.twoSum(nums, target);
    std::cout << "Test 1: ";
    printVector(result);
    
    return 0;
}

File: 01.Python/0167.py
from typing import List

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1

        while left < right:
            result_sum = numbers[left] + numbers[right]

            if result_sum < target:
                left += 1
            elif result_sum > target:
                right -= 1
            else:
                return [left + 1, right + 1]
        
        return [-1, -1]

if __name__ == "__main__":
    sol = Solution()

    numbers = [2,7,11,15]
    target = 9
    print(f"Test 1: {sol.twoSum(numbers, target)}") # Expected: [1, 2]